src/Plugin/EntityBrowser/SelectionDisplay/NewDisplay.php

```diff
@@ -0,0 +1,162 @@
+  public function getForm(array &$original_form, FormStateInterface $form_state) {
+    $form = [];
+    $form['#attached']['library'][] = 'entity_browser/new_display';
+
+    $selected_entities = $form_state->get(['entity_browser', 'selected_entities']);
+    $ids = array_keys($selected_entities);
+
+    $hidden_id = Html::getUniqueId('edit-selected-target-id');
+    $details_id = Html::getUniqueId('edit-selected');
+
+    $form += [
+      '#id' => $details_id,
+      '#type' => 'details',
+      '#open' => !empty($ids),
+      'target_id' => [
+        '#type' => 'hidden',
+        '#id' => $hidden_id,
+        // We need to repeat ID here as it is otherwise skipped when rendering.
+        '#attributes' => ['id' => $hidden_id],
+        '#default_value' => $ids,
+        // #ajax is officially not supported for hidden elements but if we
+        // specify event manually it works.
+        '#ajax' => [
+          'callback' => [get_class($this), 'updateWidgetCallback'],
+          'wrapper' => $details_id,
+          'event' => 'entity_browser_selected_updated',
+        ],
+      ],
+    ];
+
+    $form['current'] = [
+      '#theme_wrappers' => ['container'],
+      '#attributes' => ['class' => ['selected-entities-list']],
+      'items' => array_map(
+        function($id) use ($selected_entities, $details_id) {
+          $entity = $selected_entities[$id];
+          return [
+            '#theme_wrappers' => ['container'],
+            '#attributes' => [
+              'class' => ['selected-item-container'],
+              'data-entity-id' => $entity->id()
+            ],
+            'display' => ['#markup' => $entity->label()],
+            'remove_button' => [
+              '#type' => 'submit',
+              '#value' => $this->t('Remove'),
+              '#ajax' => [
+                'callback' => [get_class($this), 'updateWidgetCallback'],
+                'wrapper' => $details_id,
+              ],
+              '#submit' => [[get_class($this), 'removeItemSubmit']],
+              '#name' => 'remove_' . $id,
+              '#attributes' => ['data-entity-id' => $id]
+            ],
+          ];
+        },
+        $ids
+      ),
+    ];
+    $form['use_selected'] = array(
+      '#type' => 'submit',
+      '#value' => t('Use selected'),
+      '#name' => 'use_selected',
+    );
+
+    return $form;
+  }
+
+  /**
+   * Submit callback for remove buttons.
+   */
+  public static function removeItemSubmit(&$form, FormStateInterface $form_state) {
+    $triggering_element = $form_state->getTriggeringElement();
+    if (!empty($triggering_element['#attributes']['data-entity-id'])) {
+      $id = $triggering_element['#attributes']['data-entity-id'];
+      $parents = array_slice($triggering_element['#parents'], 0, -4);
+      $array_parents = array_slice($triggering_element['#array_parents'], 0, -4);
+
+      // Find and remove correct entity.
+      $values = explode(' ', $form_state->getValue(array_merge($parents, ['target_id'])));
```

Comment:
This will need to operate on entities that are saved in form_state, not on this hidden field (which should be removed).